### PR TITLE
Get thor mem fix 0.14.1

### PR DESCRIPTION
**Pull request by datamacgyver**
Code changes as discussed, we've reviewed at my machine so feel free to wave through ;)

All tests green

### CODE REVIEW

**Review by cqkh42 — CHANGES_REQUESTED** [review]

File: hpycc/save.py [hunk 1]
```diff
@@ -46,13 +62,15 @@ def save_output(connection, script, path_or_buf=None, syntax_check=True,
     return result.to_csv(path_or_buf=path_or_buf, **kwargs)
 
 
-def save_outputs(connection, script, directory=".", filenames=None,
-                 prefix=None, syntax_check=True, delete_workunit=True,
+def save_outputs(connection, script, directory=".", overwrite=True,
```
[the remark this answers is not in the document]
Comment (datamacgyver) [reply hpycc/save.py]: Hadn't disabled everything. Done now.

File: hpycc/connection.py [hunk 2]
```diff
@@ -21,20 +21,39 @@
...
+from urllib import parse
+from json import JSONDecodeError
+from simplejson.errors import JSONDecodeError as simpleJSONDecodeError
+from math import ceil
 
 from hpycc.utils.parsers import parse_wuid_from_failed_response, \
     parse_wuid_from_xml
 from hpycc import delete
 
+lock = threading.Lock()
```
[the remark this answers is not in the document]
Comment (datamacgyver) [reply hpycc/connection.py]: Not any more. Removed.

File: README.rst [hunk 3]
```diff
@@ -34,47 +38,128 @@ The package itself mainly uses core Python, Pandas is needed for outputting data
 
 There is a dependency for client tools to run ECL scripts (you need ecl.exe and eclcc.exe).
 Make sure you install the right client tools for your HPCC version and add the dir to your system path,
-e.g. C:\Program Files (x86)\HPCCSystems\X.X.X\clienttools\bin.
+e.g. C:\\Program Files (x86)\\HPCCSystems\\X.X.X\\clienttools\\bin.
 
-Tests require docker to spin up HPCC test environments.
+Tests and docker contianer functions require docker to spin up HPCC environments.
```
[the remark this answers is not in the document]
Comment (datamacgyver) [reply README.rst]: done

**Comment by datamacgyver** [comment]
Changes pushed. Feel free to merge when ready.

**Pull request merged**